docs(nextjs13) updated documentation for stable app dir and server components (#14284)
supabase/supabase: apps/docs/pages/guides/auth/auth-helpers/nextjs.mdx

- Node.js: `^10.13.0 || >=12.0.0`
- Next.js: `>=10`

> > Note: In [Next.js 13.4](https://nextjs.org/blog/next-13-4), The `app` directory and Server Components have reached stable status. Therefore, the new features are no longer in beta. You can refer to our experimental guide on utilizing [Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).

Reviewer: ```suggestion
> Note: As of [Next.js 13.4](https://nextjs.org/blog/next-13-4), the `app` directory and Server Components have reached stable status. Check out our guide on using [Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).
```

> > Note: In [Next.js 13.4](https://nextjs.org/blog/next-13-4), The `app` directory and Server Components have reached stable status. Therefore, the new features are no longer in beta. You can refer to our experimental guide on utilizing [Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).

Reviewer: ```suggestion
> Note: As of [Next.js 13.4](https://nextjs.org/blog/next-13-4), the `app` directory and Server Components have reached stable status. Check out our guide on using [Auth Helpers with Next.js Server Components](/docs/guides/auth/auth-helpers/nextjs-server-components).
```
